pywinauto/application.py

```diff
@@ -839,6 +841,15 @@ def __connect(self, **kwargs):
     def connect(self, **kwargs):
         """Connects to an already running process"""
         connected = False
+
+        if 'timeout' in kwargs:
+            try:
+                timeout = float(kwargs['timeout'])
+                del kwargs['timeout']
+                wait_until_passes(self.timeout, Timings.app_connect_retry, partial(self.Connect, **kwargs), (ValueError))

```

Comment:
I don't understand it. ```timeout``` argument is stored into a local variable and deleted from ```kwargs``` but the actual call ```wait_until_passes``` still uses ```self.timeout``` that is only initiated to ```0``` and never changed ? How it supposed to work. Where is a test to check that the actual timeout period has been expired and the exception has been raised ? Am I missing something ?